src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3ResponseService.java

```diff
@@ -185,4 +204,156 @@ private static String getRespondentNames(CaseData caseData) {
+     *     <li>Fetches the case user role assignments for the given case.</li>
+     *     <li>Parses solicitor roles from the case user roles and collects their indices.</li>
+     * </ul>
+     *
+     * @param userToken the authentication token of the user
+     * @param caseId the ID of the case to check roles against
+     * @return a list of integer indices indicating which respondents the user represents
+     * @throws GenericServiceException if the user or required data (e.g., user details, case roles)
+     *         cannot be found or parsed
+     * @throws IOException if an error occurs during communication with downstream services
+     */
+    public List<Integer> getRepresentedRespondentIndexes(String userToken, String caseId)
+            throws GenericServiceException, IOException {
+        checkUserTokenAndCaseid(userToken, caseId);
+
+        UserDetails userDetails = userIdamService.getUserDetails(userToken);
+        if (ObjectUtils.isEmpty(userDetails)) {
+            throw new GenericServiceException(USER_NOT_FOUND, new Exception(USER_NOT_FOUND),
+                    USER_NOT_FOUND, caseId, "Et3ResponseService", "isRespondentRepresentative");
+        }
+        if (StringUtils.isBlank(userDetails.getUid())) {
+            throw new GenericServiceException(USER_ID_NOT_FOUND, new Exception(USER_ID_NOT_FOUND),
+                    USER_ID_NOT_FOUND, caseId, "Et3ResponseService", "isRespondentRepresentative");
+        }
+
+        CaseUserAssignmentData caseUserAssignmentData = ccdCaseAssignment.getCaseUserRoles(caseId);
+        if (ObjectUtils.isEmpty(caseUserAssignmentData)) {
+            throw new GenericServiceException(CASE_ROLES_NOT_FOUND, new Exception(CASE_ROLES_NOT_FOUND),
+                    CASE_ROLES_NOT_FOUND, caseId, "Et3ResponseService", "isRespondentRepresentative");
+        }
+
+        List<Integer> solicitorIndexList = new ArrayList<>();
+        for (CaseUserAssignment caseUserAssignment : caseUserAssignmentData.getCaseUserAssignments()) {
+            if (userDetails.getUid().equals(caseUserAssignment.getUserId())) {
+                SolicitorRole solicitorRole = SolicitorRole.from(caseUserAssignment.getCaseRole()).orElseThrow();
+                solicitorIndexList.add(solicitorRole.getIndex());
+            }
+        }
+        return solicitorIndexList;
+    }
+
+    private static void checkUserTokenAndCaseid(String userToken, String caseId) throws GenericServiceException {
+        if (StringUtils.isBlank(userToken)) {
+            throw new GenericServiceException(INVALID_USER_TOKEN, new Exception(INVALID_USER_TOKEN),
+                    INVALID_USER_TOKEN, StringUtils.EMPTY, "Et3ResponseService", "isRespondentRepresentative");
+        }
+        if (StringUtils.isBlank(caseId)) {
+            throw new GenericServiceException(INVALID_CASE_ID, new Exception(INVALID_CASE_ID),
+                    INVALID_CASE_ID, caseId, "Et3ResponseService", "isRespondentRepresentative");
```

Comment:
I know we get line numbers in logs but looking at the usage of GenericServiceException, the 5th parameter is the method name. In all of these instances, isRespondentRepresentative is used which might confuse someone if they see this in the logs and it show's the incorrect method name